src/Game.js

```diff
@@ -499,36 +270,31 @@ Game.prototype = {
 		throw "lcdgames.js - "+arguments.callee.caller.toString()+", no frame with filename '" + name + "'";
 	},
 
-	setShapeByName: function(filename, value) {
-		// if called too soon
-		if (this.gamedata.frames) {
-			// find shape
-			for (var i = 0; i < this.gamedata.frames.length; i++) {
-				if (this.gamedata.frames[i].filename == filename) {
-					this.gamedata.frames[i].value = value;
-					this._refresh = true;
-					return true;
-				}
-			}
-			throw "lcdgames.js - "+arguments.callee.caller.toString()+", no frame with filename '" + filename + "'";
-		}
-		return false;
-	},
-
 	/**
-	 * Set value of Shape by Index.
+	 * Toggle shape visibility by its name.
 	 *
-	 * @private
-	 * @param {number} idx
+	 * @param {[string | number ]} filename - index or name of Shape.
 	 * @param {boolean} value
-	 * @returns {boolean}
 	 */
-	setShapeByIdx: function(idx, value) {
-		this.gamedata.frames[idx].value = value;
-		this._refresh = true;
-		return true;
+	setShapeByName: function(filename, value) {
+		let name = filename;
+		let frame;
+		// Some code still uses indexes. Kept for compatibility.
```

Comment:
Rather than refactoring the sequences code as well, I just added this hack. Really would like to hear your thoughts before doing more work.